### PR TITLE
resource_manager: implement the metrics info collecting

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5854.

### What is changed and how does it work?

```commit-message
Collect the resource group consumption info and use Prometheus to observe it.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by CabinfeverB — COMMENTED** [review]
We can send consumption to `consumptionDispatcher ` passing by `AcquireTokenBuckets`

File: pkg/mcs/resource_manager/server/grpc_service.go [hunk 1]
```diff
@@ -145,9 +145,16 @@ func (s *Service) AcquireTokenBuckets(stream rmpb.ResourceManager_AcquireTokenBu
 		targetPeriodMs := request.GetTargetRequestPeriodMs()
 		resps := &rmpb.TokenBucketsResponse{}
 		for _, req := range request.Requests {
-			rg := s.manager.GetMutableResourceGroup(req.ResourceGroupName)
+			resourceGroupName := req.GetResourceGroupName()
+			// Send the consumption to update the metrics.
+			s.manager.consumptionDispatcher <- struct {
```
Comment (CabinfeverB) [inline pkg/mcs/resource_manager/server/grpc_service.go:150]: Why not handle it after L155

**Review by CabinfeverB — COMMENTED** [review]
rest LGTM

File: pkg/mcs/resource_manager/server/grpc_service.go (hunk 1 above)
Comment (JmPotato) [reply pkg/mcs/resource_manager/server/grpc_service.go]: Fixed.

**Review by nolouch — APPROVED** [review]
lgtm

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**